in-progress/8025-publishing-tx-objects/design.md

```diff
@@ -48,27 +69,280 @@ To the DA oracle contract, the proposer will publish the following data:
   - Tx2
   - ...
 
+> Most of the global variables could be directly populated by L1, but providing it makes it a lot clearer what the reason of failure is if such a one is encountered. Also the `inHash` can be populated by L1.
 
-To the rollup contract, the proposer will publish:
+#### DA Oracle Changes
 
-- CallData
-  - numTxs
-  - txsHash (a commitment to first nullifiers of all transactions in the (eventual) block)
-  - inHash
-  - GlobalVariables
-    - slotNumber
-    - timestamp
-    - coinbase
-    - feeRecipient
+When the data is to be submitted as part of the same transaction as the block proposal to the L1 contract, we have two variations possible:
 
-#### DA Oracle Changes
+1. Remove the data availability oracle in its whole, and simply use the versioned KZG hashes directly
+2. Save the versioned KZG hashes in the data availability oracle
 
-Keeps track of the versioned KZG hashes that were present in a transaction.
+Solution 1 is significantly cheaper as it don't update storage. Furthermore, by requiring that the versioned KZG hashes are part of the proposal provided, we can show the exact KZG hashes using just the transaction and the commitment to the proposal.
 
 #### Rollup Contract Changes
 
+```python
+struct ChainTip:
+  block_number: uint256
+  slot_number: uint256
+
+
+struct State:
+  pending_tip: ChainTip
+  proven_tip: ChainTip
+
+
+struct ProposalLog:
+  hash: bytes32
+  slot_number: uint256
+  archive: bytes32
+
+
+struct GasSettings:
+  feePerDaGas: uint256
+  feePerL2Gas: uint256
+
+
+struct GlobalVariables:
+  block_number: uint256
+  slot_number: uint256
+  timestamp: uint256
+  coinbase: address
+  fee_recipient: AztecAddress # discuss
+  gas_settings: GasSettings
+
+
+struct Attestations:
+  signature: AggregateBLS
+  missing: uint256 # support up to 3*256 members in committee
```

Comment:
Oh, this is only if we can change the ordering such that we have the "missing" at the start. Which likely will cause of a lit more pain, so it is likely much simpler to just store commitee_size / 256 uints for bitmap flags instead.